packages/lesswrong/components/surveys/SurveyPostsItem.tsx

```diff
@@ -0,0 +1,382 @@
+        itemType: "survey",
+        surveyId: survey._id,
+      });
+      return newOpen;
+    });
+  }, [captureEvent, survey._id]);
+
+  const {create: createResponse} = useCreate({
+    collectionName: "SurveyResponses",
+    fragmentName: "SurveyResponseMinimumInfo",
+  });
+
+  const onSubmit = useCallback(async (response: Record<string, QuestionResponse>) => {
+    setIsSubmitting(true);
+    try {
+      await createResponse({
+        data: {
+          surveyId: survey._id,
+          surveyScheduleId,
+          userId: currentUser?._id,
+          clientId,
+          response,
+        },
+      });
+      captureEvent("surveySubmit", {
+        surveyId: survey._id,
+        surveyScheduleId,
+        response,
+      });
+      void refetchSurvey?.();
+      collapse();
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error("Error submitting survey:", e);
+      setIsSubmitting(false);
+    }
+  }, [
+    collapse,
+    refetchSurvey,
+    captureEvent,
+    survey._id,
+    surveyScheduleId,
+    createResponse,
+    currentUser?._id,
+    clientId,
+  ]);
+
+  const onRespond = useCallback(async (
+    questionId: string,
+    response: QuestionResponse,
+  ) => {
+    setResponses((responses) => {
+      const newResponses = {
+        ...responses,
+        [questionId]: response,
+      };
+      if (Object.keys(newResponses).length === survey.questions.length) {
+        void onSubmit(newResponses);
+      }
+      return newResponses;
+    });
+    setResponses((responses) => ({
+      ...responses,
+      [questionId]: response,
+    }));
+    captureEvent("surveyResponse", {
+      surveyId: survey._id,
+      surveyScheduleId,
+      questionId,
+      response,
+    });
+  }, [captureEvent, survey, surveyScheduleId, onSubmit]);
+
+  const onDismiss = useCallback(() => {
+    collapse();
+    setTimeout(() => {
+      setCookie(
+        HIDE_SURVEY_SCHEDULE_IDS,
+        [...(hideSurveyScheduleIds || []), surveyScheduleId],
+      );
+    }, 500);
+    captureEvent("surveyDismiss", {
+      surveyId: survey._id,
+      surveyScheduleId,
+    });
+  }, [
+    collapse,
+    captureEvent,
+    survey._id,
+    surveyScheduleId,
+    hideSurveyScheduleIds,
+    setCookie,
+  ]);
+
+  const onOptOut = useCallback(() => {
+    void updateCurrentUser({optedOutOfSurveys: true});
+    collapse();
+    void refetchSurvey?.();
+    captureEvent("surveyOptOut", {
+      surveyId: survey._id,
+      surveyScheduleId,
+    });
+  }, [
+    captureEvent,
+    survey._id,
+    surveyScheduleId,
+    refetchSurvey,
+    collapse,
+    updateCurrentUser,
+  ]);
+
+  if (hideSurveyScheduleIds?.includes(surveyScheduleId)) {
+    return null;
+  }
+
+  const {
+    LWTooltip, ForumIcon, LWClickAwayListener, DropdownMenu, DropdownItem,
+    PopperCard, Loading,
+  } = Components;
+  return (
+    <div className={classes.root}>
+      <LWTooltip
+        title="Some really informative stuff"
```

Comment:
TODO: Add some informative tooltip text here (maybe we want to link to an explanatory quick take or something?)